hlf/chaincode/admission.md

```diff
@@ -0,0 +1,227 @@
+# Hyperledger Admission Api
+
+In the following, the chaincode api for the "Admission"-Chaincode is described.
+This contains all Transactions and Models provided for this domain.
+The Errors returned are defined [here](errors.md#Errors).
+
+## Transactions
+
+### Add Admission
+- ID = addAdmission
+- Send
+    - enrollmentId :: String
+    - courseId :: String
+    - moduleId :: String
+    - timestamp :: String \<DATE YYYY-MM-DD_hh:mm\>
+- Receive
+    - [Admission](#Admission)
+      -  Description: Done, returns the submitted data, decorated with an admissionId.
+
+    - [DetailedError](errors.md#DetailedError) 
+      ```json
+      {
+        "type": "HLUnprocessableEntity",
+        "title": "The following parameters do not conform to the specified format",
+        "invalidParams": [
+          {
+            "name": "string",
+            "reason": "string"
+          }
+        ]
+      }
+      ```
+       - Description: This error is returned, if the given parameters could not be parsed. If some attributes within the json are not well formatted, they are listed in invalidParams.  
+       For detailed informations see [parameter checks](#parameterChecks).
+
+    - [SemanticError](errors.md#SemanticError)
+      ```json
+      {
+        "type": "HLInvalid",
+        "title": "Semantic errors were found. The following Validation Rules were violated:",
+        "validationRules": [
+          {
+            "name": "ModuleAccess",
+            "reason": "Student is not matriculated in an examinationRegulation containing the refernced module."
+          }
+        ]
+      }
+      ```
+       - Description: This error is returned, if the Student is not allowed to be admissioned in this way.
+         For detailed informations see [semantic checks](#semanticChecks).

```

Comment:
Why is that a new error? That sounds a lot like a validation error. I would propose to just use the HLUnprocessableEntity, since vue will receive a 422 either way that is not an incorrect error

Reply:
The name UnprocessableEntity itself is a limitation. It suggest some error with the format / syntax.
To extend this we need to add new errors or change all existing ones.
Adding a new one to distinguish between syntactical and semantic errors seems easier to me.

The Semantic Error was a garbage idea though, we can perfectly map semantic errors to the existing "DetailedError"-code